### PR TITLE
Support for publishing events over RabbitMQ when TruckDriveStates are created

### LINKED ISSUE #77: Publish events when truckdrivestates are created

### PR DESCRIPTION
**Pull request by villejuutila**
- Also renamed generated packages to better support multi modules
- Added support for tests MessagingClient to consume messages

Resolves #77

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/vp/vehiclemanagement/trucks/TrucksApiImpl.kt [hunk 1]
```diff
@@ -281,7 +286,10 @@ class TrucksApiImpl: TrucksApi, AbstractApi() {
         if (requestApiKey != apiKey) return@withCoroutineScope createForbidden(INVALID_API_KEY)
         val truck = truckController.findTruck(truckId) ?: return@withCoroutineScope createNotFound(createNotFoundMessage(TRUCK, truckId))
 
-        truckDriveStateController.createDriveState(truck, truckDriveState) ?: return@withCoroutineScope createAccepted(null)
+        val createdDriveState = truckDriveStateController.createDriveState(truck, truckDriveState) ?: return@withCoroutineScope createAccepted(null)
+
+        eventBus.publish(TruckDriveStateCreatedConsumer.TRUCK_DRIVE_STATE_CREATED, createdDriveState)
```
Comment (anttileppa) [inline src/main/kotlin/fi/metatavu/vp/vehiclemanagement/trucks/TrucksApiImpl.kt:291]: Is there any particular reason why this is not in the controller?

File: src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt [hunk 2]
```diff
@@ -12,6 +12,16 @@ class DefaultTestProfile: QuarkusTestProfile {
     override fun getConfigOverrides(): MutableMap<String, String> {
         val config: MutableMap<String, String> = HashMap()
         config["vp.vehiclemanagement.telematics.apiKey"] = VEHICLE_MANAGEMENT_TELEMATICS_API_KEY
+
+        config["mp.messaging.incoming.vp-in.connector"] = "smallrye-rabbitmq"
```
Comment (anttileppa) [inline src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt:16]: Isn't it always rabbitmq? If it is then why this is defined in the test, not in the application.properties

File: src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt [hunk 3]
```diff
@@ -12,6 +12,16 @@ class DefaultTestProfile: QuarkusTestProfile {
...
+//        config["mp.messaging.incoming.vp-in.queue.name"] = "incoming_queue"
```
Comment (anttileppa) [inline src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt:17]: Remove commented code

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/vp/vehiclemanagement/trucks/TrucksApiImpl.kt (hunk 1 above)
Comment (villejuutila) [reply src/main/kotlin/fi/metatavu/vp/vehiclemanagement/trucks/TrucksApiImpl.kt]: Not really. Moved there if that's more appropriate

File: src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt (hunk 3 above)
Comment (villejuutila) [reply src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt:17]: Resolved

File: src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt (hunk 2 above)
Comment (villejuutila) [reply src/test/kotlin/fi/metatavu/vp/vehiclemanagement/test/functional/settings/DefaultTestProfile.kt]: Good point. Moved.

**Review by anttileppa — APPROVED** [review]

**Pull request merged**